notebooks/learning-to-rank/01-learning-to-rank.ipynb

```diff
@@ -0,0 +1,1556 @@
+              "...          ...                  ...     ...    ...         ...          ...   \n",
+              "384750  qid:3383   2013 the wolverine  263115      0         NaN          NaN   \n",
+              "384751  qid:3383   2013 the wolverine   25913      0         NaN          NaN   \n",
+              "384752  qid:3383   2013 the wolverine  567604      0         NaN          NaN   \n",
+              "384753  qid:3383   2013 the wolverine  533535      0         NaN          NaN   \n",
+              "384754  qid:3383   2013 the wolverine  876327      0         NaN          NaN   \n",
+              "\n",
+              "        title_all_terms_bm25  popularity  \n",
+              "0                        NaN      13.628  \n",
+              "1                        NaN      64.003  \n",
+              "2                        NaN     143.211  \n",
+              "3                        NaN      32.913  \n",
+              "4                        NaN      21.058  \n",
+              "...                      ...         ...  \n",
+              "384750                   NaN      68.287  \n",
+              "384751                   NaN      21.026  \n",
+              "384752                   NaN      32.913  \n",
+              "384753                   NaN      34.773  \n",
+              "384754                   NaN      25.920  \n",
+              "\n",
+              "[384755 rows x 8 columns]"
+            ]
+          },
+          "execution_count": 10,
+          "metadata": {},
+          "output_type": "execute_result"
+        }
+      ],
+      "source": [
+        "import numpy\n",
+        "\n",
+        "from eland.ml.ltr import FeatureLogger\n",
+        "\n",
+        "# First we create a feature logger that will be used to query Elasticsearch to retrieve the features:\n",
+        "feature_logger = FeatureLogger(es_client, MOVIE_INDEX, ltr_config)\n",
+        "\n",
+        "# This method will be applied for each group of query in the judgment log:\n",
+        "def _extract_query_features(query_judgements_group):\n",
+        "    # Retrieve document ids in the query group as strings.\n",
+        "    doc_ids = query_judgements_group['doc_id'].astype('str').to_list()\n",
+        "\n",
+        "    # Resolve query paras for the current query group (e.g.: {\"query\": \"batman\"}).\n",
+        "    query_params = { 'query': query_judgements_group['query'].iloc[0] }\n",
+        "\n",
+        "    # Extract the features for the documents in the query group:\n",
+        "    doc_features =  feature_logger.extract_features(query_params, doc_ids)\n",
+        "\n",
+        "    # Adding a column to the dataframe for each features:\n",
+        "    for feature_index, feature_name in enumerate(feature_logger._model_config.feature_names):\n",
+        "        query_judgements_group[feature_name] = numpy.array([doc_features[doc_id][feature_index] for doc_id in doc_ids])\n",
```

Comment:
Because `FeatureLogger.extract_features` is already released we can not change it anymore. So we have to deal with this design. 

I did remove the `feature_logger._model_config.feature_names` to replace it with `ltr_config.feature_names`

In the future, I would like to add a method to the feature logger that accept a data frame and few specs about the column and would return the data frame with the features.

Reply:
I resolve the comment since there is nothing more that can be done do in the context of this PR.